### ISSUE TITLE
Question about comparability -

### ISSUE THREAD

**Issue by Sharing-Sam-Work** [flairNLP/flair#2676]
### Problem
Given a same model, a same corpus, and a trainer object instanciated from the two, I find during evaluation that training gives inconsistent results. 
For comparability, I would like to know how to control randomness in training. Is there an equivalent of the random seed parameter used in transformer?
So far, I have tried saving a model and a corpus to use them twice, to avoid randomness due to model instanciation.:

### Saving  a model:
      embedding_types = [
      
          WordEmbeddings('glove'),
      
         
          FlairEmbeddings('news-forward'),
          FlairEmbeddings('news-backward'),
      ]
      
      embeddings = StackedEmbeddings(embeddings=embedding_types)
      
      tagger = SequenceTagger(hidden_size=256,
                          embeddings=embeddings,
                          tag_dictionary=label_dict,
                          tag_type=label_type,
                          use_crf=True)
      tagger.save("base_model")`

### Saving a corpus:
        # 1. get the corpus 
        # define columns
        columns = {0: 'text', 1: 'ner'}
        
        # this is the folder in which train, test and dev files reside
        data_folder = path_data
        
        # init a corpus using column format, data folder and the names of the train, dev and test files
        corpus: Corpus = ColumnCorpus(data_folder, columns,
                                      train_file='train.txt',
                                      test_file='test.txt')
            
        corpus=corpus.downsample(0.1)
        # 2. what label do we want to predict?
        label_type = 'ner'
        
        # 3. make the label dictionary from the corpus
        label_dict = corpus.make_label_dictionary(label_type=label_type)
        print(label_dict)
        

The two following trainings (same code executed in different cells of a jupyter notebook) yield different results:
### Test 1:
          `tagger=SequenceTagger.load("base_model")
          trainer = ModelTrainer(tagger, corpus)
          # 7. start training
          trainer.train('resources/taggers/tests/1',
                        learning_rate=0.1,
                        mini_batch_size=32,
                        write_weights = True,
                        max_epochs=150)`
### Test 2:
      `tagger=SequenceTagger.load("base_model")
      trainer = ModelTrainer(tagger, corpus)
      # 7. start training
        trainer.train('resources/taggers/tests/2',
                      learning_rate=0.1,
                      mini_batch_size=32,
                      write_weights = True,
                      max_epochs=150)`

I have also tried playing with the ModelTrainer train method parameters (shuffle=False), but it didn't work.

### How to achieve comparability?
If there is no way to operate that allows result consistency, it introduces uncertainty as I don't know what prompts a difference in performance. 
What do you recommend for comparability in the case of two different models? 
If I want to compare them, should I train each model 5 times and take the average of the results to obtain a fair comparison?
Thank you!

**Comment by alanakbik** [comment]
Hello, @Sharing-Sam-Work yes there is a direct equivalent. Set 

```python
import flair
flair.set_seed(123)
```

to get reproducible results!

**Comment by Sharing-Sam-Work** [comment]
Thank you very much for the quick answer ! :)

**Comment by Sharing-Sam-Work** [comment]
Hello, 
I tried setting the seed, but I still get different results for the code above (max epoch = 2)

##### Result 1
          {'test_score': 0.0,
           'dev_score_history': [0.06741573033707865, 0.16129032258064516],
           'train_loss_history': [3.247536113029466, 2.3094709477152393],
           'dev_loss_history': [tensor(2.1150, device='cuda:0'),
            tensor(0.9884, device='cuda:0')]}
##### Result 2
          {'test_score': 0.0,
           'dev_score_history': [0.07462686567164178, 0.1935483870967742],
           'train_loss_history': [3.2399936064830803, 2.2099512707901705],
           'dev_loss_history': [tensor(2.1079, device='cuda:0'),
            tensor(0.9660, device='cuda:0')]}
Is it expected behaviour?

**Comment by Sharing-Sam-Work** [comment]
Hello, 
I tried setting the seed, but I still get different results for the code above (max epoch = 2)

##### Result 1
          {'test_score': 0.0,
           'dev_score_history': [0.06741573033707865, 0.16129032258064516],
           'train_loss_history': [3.247536113029466, 2.3094709477152393],
           'dev_loss_history': [tensor(2.1150, device='cuda:0'),
            tensor(0.9884, device='cuda:0')]}
##### Result 2
          {'test_score': 0.0,
           'dev_score_history': [0.07462686567164178, 0.1935483870967742],
           'train_loss_history': [3.2399936064830803, 2.2099512707901705],
           'dev_loss_history': [tensor(2.1079, device='cuda:0'),
            tensor(0.9660, device='cuda:0')]}
Is it expected behaviour?

**Comment by alanakbik** [comment]
You need to re-set the seed for each experiment individually. For instance, when I do a parameter sweep with a for-loop, I set the seed at the beginning of each experiment (before the model is initialized).

**Comment by Sharing-Sam-Work** [comment]
Thank you it worked!
